koa-crud/src/pagination/paginate.ts

```diff
@@ -0,0 +1,48 @@
+import rType from 'ramda';
+import { Connection } from '../types';
+
+const makePaginate = ({ R }: { R: typeof rType }) => {
+  return function <T extends { cursorBuffer: Buffer }>({
+    data,
+    first,
+    after,
+  }: {
+    data: T[];
+    first: number;
+    after: string;
+  }): Connection<T> {
+    if (first < 0) {
+      throw new Error(`Invalid first`);
+    }
+
+    const startingIndex = R.findIndex(
+      R.propEq('cursorBuffer', Buffer.from(after, 'base64')),
+    )(data);
+
+    if (startingIndex < 0) {
+      throw new Error(`Invalid cursor`);
+    }
```

Comment:
Please make a custom error for invalid cursor

Reply:
noted sir